File: Convex_Optimization_I_Boyd/spacecraft_landing_data.jl
h = 1;
g = 0.1;
m = 10;
Fmax = 10;
p0 = [50,50,100];
v0 = [-10,0,-10];
alpha = 0.5;
gamma = 1;
K = 35;

# use the following code to plot your trajectories
# plot the glide cone (don't modify)
# -------------------------------------------------------
# using PyPlot
# x = linspace(-40,55,30); y = linspace(0,55,30);
# X = repmat(x', length(x), 1);
# Y = repmat(y, 1, length(y));
# Z = alpha*sqrt(X.^2+Y.^2);
# figure();
# grid(true);
# hold(true);
# surf(X, Y, Z, cmap=get_cmap("autumn"));
# xlim([-40, 55]);
# ylim([0, 55]);
# zlim([0, 105]);

# INSERT YOUR VARIABLES HERE:
# -------------------------------------------------------
# Make sure you pass plot and quiver column vectors and
# not row vectors. To plot 3D quiver plots you need to be
# on matplotlib 1.4. Quiver plots in matplotlib are also
# specified by the location of the arrow HEADS and the
# direction, instead of the arrow base
#
# plot(p[:,1], p[:,2], p[:,3]);
# heads = p[1:K,:] + f;
# quiver(heads[:,1], heads[:,2], heads[:,3],
#   f[:,1], f[:,2], f[:,3], length=10);

File: Convex_Optimization_I_Boyd/spacecraft_landing_data.py
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from matplotlib import cm

h = 1.
g = 0.1
m = 10.
Fmax = 10.
p0 = np.matrix('50 ;50; 100')
v0 = np.matrix('-10; 0; -10')
alpha = 0.5
gamma = 1.
K = 35

# use the following code to plot your trajectories
# and the glide cone (don't modify)
# -------------------------------------------------------
fig = plt.figure()
ax = fig.gca(projection='3d')

X = np.linspace(-40, 55, num=30)
Y = np.linspace(0, 55, num=30)
X, Y = np.meshgrid(X, Y)
Z = alpha*np.sqrt(X**2+Y**2)
ax.plot_surface(X, Y, Z, rstride=1, cstride=1, cmap=cm.coolwarm, linewidth=0)
#Have your solution be stored in p
#ax.plot(xs=p.value[0,:].A1,ys=p.value[1,:].A1,zs=p.value[2,:].A1)
ax.set_xlabel('x'); ax.set_ylabel('y'); ax.set_zlabel('z')
plt.show()
